internal/compiler/widgets/fluent-base/std-widgets-base.slint

```diff
@@ -519,19 +523,38 @@ component StandardListViewBase inherits ListView {
         touch := TouchArea {
             width: parent.width;
             height: parent.height;
-            clicked => { root.current-item = idx; }
+
+            clicked => { 
+                set-current-item(idx);    
+            }
+        }
+    }
+
+    public function set-current-item(index: int) {
+        if(index < 0 || index >= model.length) {
+            return;
+        }
+
+        root.current-item = index;
+
+        if(current-item-y < 0) {
+            self.viewport-y += 0 - current-item-y;
+        }
+        
+        if(current-item-y + item-height > self.visible-height) {
+            self.viewport-y -= current-item-y + item-height - self.visible-height;
```

Comment:
I think you can omit `root` and `self` qualifiers here. Another option would be to consistently use `self`. A matter of taste I guess :-)

```suggestion
        current-item = index;

        if(current-item-y < 0) {
            viewport-y += 0 - current-item-y;
        }
        
        if(current-item-y + item-height > visible-height) {
            viewport-y -= current-item-y + item-height - visible-height;
```